cmd/prepare-engine.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"github.com/go-flutter-desktop/hover/internal/build"
+	"github.com/go-flutter-desktop/hover/internal/config"
+	"github.com/go-flutter-desktop/hover/internal/enginecache"
+	"github.com/go-flutter-desktop/hover/internal/log"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"runtime"
+)
+
+var (
+	prepareCachePath     string
+	prepareEngineVersion string
+	prepareReleaseMode   bool
+	prepareDebugMode     bool
+	prepareBuildModes    []build.Mode
+)
+
+func init() {
+	prepareEngineCmd.PersistentFlags().StringVar(&prepareCachePath, "cache-path", enginecache.DefaultCachePath(), "The path that hover uses to cache dependencies such as the Flutter engine .so/.dll")
+	prepareEngineCmd.PersistentFlags().StringVar(&prepareEngineVersion, "engine-version", config.BuildEngineDefault, "The flutter engine version to use.")
+	prepareEngineCmd.PersistentFlags().BoolVar(&prepareDebugMode, "debug-mode", true, "Prepare the flutter engine for debug mode")
+	prepareEngineCmd.PersistentFlags().BoolVar(&prepareReleaseMode, "release-mode", false, "Prepare the flutter engine for release mode")
+	prepareEngineCmd.AddCommand(prepareEngineLinuxCmd)
+	prepareEngineCmd.AddCommand(prepareEngineDarwinCmd)
+	prepareEngineCmd.AddCommand(prepareEngineWindowsCmd)
+	rootCmd.AddCommand(prepareEngineCmd)
+}
+
+var prepareEngineCmd = &cobra.Command{
+	Use:   "prepare-engine",
+	Short: "Validates or updates the flutter engine on this machine for a given platform",
+}
+
+var prepareEngineLinuxCmd = &cobra.Command{
+	Use:   "linux",
+	Short: "Validates or updates the flutter engine on this machine for a given platform",
+	Run: func(cmd *cobra.Command, args []string) {
+		initPrepareEngineParameters("linux")
+		subcommandPrepare("linux")
+	},
+}
+
+var prepareEngineDarwinCmd = &cobra.Command{
+	Use:   "darwin",
+	Short: "Validates or updates the flutter engine on this machine for a given platform",
+	Run: func(cmd *cobra.Command, args []string) {
+		initPrepareEngineParameters("darwin")
+		subcommandPrepare("darwin")
+	},
+}
+
+var prepareEngineWindowsCmd = &cobra.Command{
+	Use:   "windows",
+	Short: "Validates or updates the flutter engine on this machine for a given platform",
+	Run: func(cmd *cobra.Command, args []string) {
+		initPrepareEngineParameters("windows")
+		subcommandPrepare("windows")
+	},
+}
+
+func initPrepareEngineParameters(targetOS string) {
+	if prepareDebugMode {
+		prepareBuildModes = append(prepareBuildModes, build.DebugMode)
+	}
+	if prepareReleaseMode {
+		prepareBuildModes = append(prepareBuildModes, build.ReleaseMode)
+	}
```

Comment:
Validate that only one of those flags is given

Reply:
I'm willing to do this. It will mean that the debug-mode default being true will need to be changed. Are you saying that you want people to run two commands when they want to prepare both debug and release engines?